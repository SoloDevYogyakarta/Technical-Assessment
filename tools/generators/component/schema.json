{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "component",
  "type": "object",
  "properties": {
    "filename": {
      "type": "string",
      "description": "Write your file name ? ",
      "x-prompt": {
        "type": "string",
        "message": "Write your filename"
      },
      "$default": {
        "$source": "filename"
      }
    },
    "directoryname": {
      "type": "string",
      "description": "Write your directory name ? ",
      "x-prompt": {
        "type": "string",
        "message": "Write your directoryname"
      },
      "$default": {
        "$source": "directoryname"
      }
    },
    "projectname": {
      "type": "string",
      "description": "Write your project name ? ",
      "x-prompt": {
        "type": "string",
        "message": "Write your projectname"
      },
      "$default": {
        "$source": "projectname"
      }
    }
  }
}
