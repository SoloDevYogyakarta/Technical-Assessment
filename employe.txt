“username”: string, “firstName”:string, “lastName”:string, “email”:string, “birthDate”:datetime, “basicSalary”:double, “status”:string, “group”:string, “description”:datetime