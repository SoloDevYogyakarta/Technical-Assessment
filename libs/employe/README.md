# employe

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test employe` to execute the unit tests.
