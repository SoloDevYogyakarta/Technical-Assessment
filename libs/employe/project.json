{
  "name": "employe",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "projectType": "library",
  "sourceRoot": "libs/employe/src",
  "prefix": "nov9-technical-assessment",
  "targets": {
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/employe/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": ["libs/employe/**/*.ts", "libs/employe/**/*.html"]
      }
    }
  },
  "tags": []
}
