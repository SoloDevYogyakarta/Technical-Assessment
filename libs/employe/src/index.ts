export * from './lib/employe.module';
