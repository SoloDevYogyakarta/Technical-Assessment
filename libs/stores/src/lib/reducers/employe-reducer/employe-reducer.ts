import { createReducer, on } from '@ngrx/store';
import { IEmployeState } from '../../constants/employe-type/employe-type';
import * as actions from '../../actions/employe-action/employe-action';

export const initialState: IEmployeState = {
  employe: {
    username: '',
    firstName: '',
    lastName: '',
    email: '',
    birthDate: '',
    basicSalary: '',
    status: '',
    group: '',
    description: '',
  },
  employes: [],
  filterEmployes: [],
  isLoading: false,
  error: '',
};

export const employeReducer = createReducer(
  initialState,
  // Get List Employe
  on(actions.getEmployes, (state) => ({
    ...state,
    isLoading: true,
  })),
  on(actions.getEmployesSuccess, (state, action) => {
    return {
      ...state,
      employes: action.employes,
      isLoading: action.isLoading,
    };
  }),
  on(actions.getEmployesFailed, (state, action) => ({
    ...state,
    isLoading: action.isLoading,
    error: action.error,
  })),
  // ======================================
  on(actions.destroyEmploye, (state, action) => ({
    ...state,
    isLoading: true,
  })),
  on(actions.destroyEmployeSuccess, (state, action) => {
    return {
      ...state,
      employes: state.employes.filter(
        (employe) => employe.username !== action.username
      ),
      isLoading: action.isLoading,
    };
  }),
  on(actions.getDetailEmployeSuccess, (state, action) => {
    return {
      ...state,
      employe: state.employes.filter(
        (employe) => employe.username === action.username
      )[0],
      isLoading: action.isLoading,
    };
  }),
  on(actions.filterEmployeSuccess, (state, action) => ({
    ...state,
    filterEmployes: state.employes.filter(
      (item) =>
        item.username.toLowerCase().indexOf(action.username.toLowerCase()) > -1
    ),
    isLoading: action.isLoading,
  }))
);
