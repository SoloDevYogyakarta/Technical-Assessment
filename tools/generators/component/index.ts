import {
  Tree,
  formatFiles,
  getProjects,
  names,
  generateFiles,
} from '@nrwl/devkit';
import { join } from 'path';
import { ISchema } from '../schema';

export default async function (tree: Tree, schema: ISchema) {
  const workspace = getProjects(tree).get(schema.projectname);
  const templatePath = 'tools/generators/component/templates';
  const directoryPath = schema.directoryname
    ? `${workspace?.sourceRoot}/lib/${schema.directoryname}`
    : `${workspace?.sourceRoot}/lib`;
  const filename = `${schema.filename}.component`;
  const targetPath = join(directoryPath, filename);
  const componentNames = names(filename);
  const substitutions = {
    tmpl: '',
    ...componentNames,
  };
  generateFiles(tree, templatePath, targetPath, substitutions);
  await formatFiles(tree);
}
