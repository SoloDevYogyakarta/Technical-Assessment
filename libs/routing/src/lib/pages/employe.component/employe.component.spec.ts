import { EmployeComponent } from './employe.component';
import {
  TestBed,
  TestBedStatic,
  ComponentFixture,
} from '@angular/core/testing';
import { StoreModule } from '@ngrx/store';
import { employeReducer } from 'libs/stores/src/lib/reducers/employe-reducer/employe-reducer';

describe('EmployeComponent', () => {
  let container: TestBedStatic;
  let component: ComponentFixture<EmployeComponent>;

  beforeEach(async () => {
    container = await TestBed.configureTestingModule({
      imports: [StoreModule.forRoot([employeReducer])],
      declarations: [EmployeComponent],
    }).compileComponents();

    component = TestBed.createComponent(EmployeComponent);
    component.detectChanges();
  });

  it('should render snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it("should have as title 'EmployeComponent'", () => {
    const template = component.componentInstance;
    expect(template.title).toEqual('EmployeComponent');
  });

  it('should render title', () => {
    const template = component.nativeElement as HTMLElement;
    expect(template.querySelector('div')?.textContent).toContain('');
  });

  afterEach(async () => {
    await jest.clearAllMocks();
  });
});
