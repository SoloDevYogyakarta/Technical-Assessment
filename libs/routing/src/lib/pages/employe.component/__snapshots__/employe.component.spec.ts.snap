// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeComponent should render snapshot 1`] = `
<employe-component
  __ngContext__="0"
  employe$={[Function Store]}
  isLoading$={[Function Store]}
  search=""
  store={[Function Store]}
  title={[Function String]}
>
  <div
    class="flex items-center justify-center w-full pt-12"
  >
    <div
      class="max-w-md w-full"
    >
      <input
        class="w-full outline-none border-[1px] border-gray-300 rounded-lg py-2 transition duration-500 hover:border-green-600 px-3 text-gray-500 mb-3"
        id="search"
        name="search"
        placeholder="Search ..."
        type="text"
      />
    </div>
  </div><div
    class="flex items-center justify-center w-full flex-wrap"
  >
  </div>
</employe-component>
`;
